misc-bench/cpp-copy/baselines/GPU/copy.cu

```diff
@@ -0,0 +1,169 @@
+/* File:     copy.cu
+ * Purpose:  Implement vector copying on a GPU
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <assert.h>
+#include <stdint.h>
+#include <iomanip>
+#include <chrono>
+#include <omp.h>
+#include <iostream>
+#include <vector>
+
+#include "utilBaselines.h"
+
+// Params ---------------------------------------------------------------------
+typedef struct Params
+{
+  uint64_t vectorLength;
+  bool shouldVerify;
+} Params;
+
+void usage()
+{
+  fprintf(stderr,
+          "\nUsage:  ./copy.out [options]"
+          "\n"
+          "\n    -l    input size (default=2048 elements)"
+          "\n    -v    t = verifies PIM output with host output. (default=false)"
+          "\n");
+}
+
+struct Params input_params(int argc, char **argv)
+{
+  struct Params p;
+  p.vectorLength = 2048;
+  p.shouldVerify = false;
+
+  int opt;
+  while ((opt = getopt(argc, argv, "h:l:v:")) >= 0)
+  {
+    switch (opt)
+    {
+    case 'h':
+      usage();
+      exit(0);
+      break;
+    case 'l':
+      p.vectorLength = strtoull(optarg, NULL, 0);
+      break;
+    case 'v':
+      p.shouldVerify = (*optarg == 't') ? true : false;
+      break;
+    default:
+      fprintf(stderr, "\nUnrecognized option!\n");
+      usage();
+      exit(0);
+    }
+  }
+  return p;
+}
+
+/**
+ * @brief gpu copy kernel
+ */
+__global__ void copy(int* gpu_x, int* gpu_y, size_t sz) {
+  size_t idx = blockIdx.x*blockDim.x + threadIdx.x;
+  if(idx < sz) {
+    gpu_y[idx] = gpu_x[idx];
+  }
+}
+
+/**
+ * @brief Main of the Host Application.
+ */
+int main(int argc, char **argv)
+{
+  struct Params params = input_params(argc, argv);
+
+  uint64_t vectorSize = params.vectorLength;
+
+  std::vector<int> X;
+  std::vector<int> Y;
+
+  getVector(vectorSize, X);
```

Comment:
If I am not mistaken, `getVector` is a template function so you need to call it as following:

```suggestion
  getVector<int>(vectorSize, X);
```

Reply:
Could you please fix this in both PRs?